src/core/schema.js

```diff
@@ -138,6 +138,7 @@ function parseProperty (value, propDefinition) {
   // Use default value if value is falsy.
   if (value === undefined || value === null || value === '') {
     value = propDefinition.default;
+    if (Array.isArray(value)) { value = value.slice(); }
```

Comment:
Can we have an `object` type in the property types that is shallow? It allows just one level where the attributes are primitive types based on the default value? Should we deal with this in a separate issue / PR?